Tests/MongoSwiftTests/DNSSeedlistTests.swift

```diff
@@ -62,21 +62,19 @@ final class DNSSeedlistTests: MongoSwiftTestCase {
             defer { center.removeObserver(observer) }
 
             // Enclose all of the potentially throwing code in `doTest`. Sometimes the expected errors come when
-            // parsing the URI, and other times they are not until we try to send a command.
-            func doTest() throws {
-                let opts = TLSOptions(
-                    allowInvalidHostnames: true,
+            // parsing the URI, and other times they are not until we try to select a server.
+            func doTest() throws -> ConnectionString {
+                let tlsOpts = TLSOptions(
                     caFile: URL(string: MongoSwiftTestCase.sslCAFilePath ?? ""),
-                    pemFile: URL(string: MongoSwiftTestCase.sslPEMKeyFilePath ?? "")
-                )
-                let client = try MongoClient(
-                    testCase.uri,
-                    options: ClientOptions(serverMonitoring: true, tlsOptions: opts)
+                    pemFile: URL(string: MongoSwiftTestCase.sslPEMKeyFilePath ?? ""),
+                    weakCertValidation: true
                 )
-
-                // mongoc connects lazily so we need to send a command.
-                let db = client.db("test")
-                _ = try db.runCommand(["isMaster": 1])
+                let opts = ClientOptions(serverMonitoring: true, tlsOptions: tlsOpts)
+                return try self.withTestClient(testCase.uri, options: opts) { client in
+                    // try selecting a server to trigger SDAM
+                    _ = try client.connectionPool.selectServer(forWrites: false)
```

Comment:
so previously we ran an `isMaster` to trigger SDAM starting and get the `TopologyDescriptionChangedEvent`s we make assertions on.

however, this approach doesn't work for the auth tests, because if you provide a connection string with a username:password in it and attempt to run an initial command (even if the command doesn't require auth), libmongoc will both start SDAM and attempt to authenticate in an initial handshake. the clusters `mongo-orchestration` starts up do not have existing users on them that match the users used in the DNS test connection strings, so we would always get an authentication error.
we *could* just ignore that error, but the failed auth attempt means the server we couldn't authenticate to gets removed from the topology description, which makes it difficult to perform the assertions about discovered hosts below.

Kevin suggested switching to using `mongoc_client_select_server` instead, as this will start up the SDAM monitoring threads and resolve TXT records, giving us the information we need below, without performing the handshake or auth.